### PR TITLE
chore: added DevRev SDK in the functions

### PR DESCRIPTION
**Pull request by altafDevRev**
### Description
Added DevRev SDK to all the functions instead of using `axios` or `node` libraries to call DevRev APIs

### Work Item
https://app.devrev.ai/devrev/works/ISS-55066

### CODE REVIEW

File: 5-custom-webhook/code/src/functions/on_work_creation/index.ts [hunk 1]
```diff
@@ -2,51 +2,25 @@
...
+    endpoint: API_BASE,
+    token: devrevPAT,
+  })
   const workCreated = event.payload.work_created;
   const bodyComment = event.payload.body;
   const body = {
     object: workCreated,
     type: 'timeline_comment',
     body: bodyComment,
   }
```
Comment (codeon) [inline 5-custom-webhook/code/src/functions/on_work_creation/index.ts:22]: this should be of type TimelineEntriesCreateRequest.

File: 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/index.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ async function EventListener(event: any) {
 	const currStage: string = event.payload.work_updated.work.stage.name;
 	const workType: string = event.payload.work_updated.work.type;
 	const snap_in_token = event.context.secrets.service_account_token;
+	const api_base = event.execution_metadata.devrev_endpoint;
```
Comment (codeon) [inline 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/index.ts:23]:
```suggestion
	const baseURL = event.execution_metadata.devrev_endpoint;
```

File: 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/index.ts [hunk 3]
```diff
@@ -29,18 +30,18 @@ async function EventListener(event: any) {
...
 		const partID = event.payload.work_updated.work.applies_to_part.id;
-		const partObject = await getPart(partID, snap_in_token);
+		const partObject = await getPart(partID, snap_in_token, api_base);
 
 		console.log(`Ticket ${ticketID} moved to Product Assist stage`);
 
 		if ((partObject.part.owned_by).length == 1 && partObject.part.owned_by[0].type != "dev_user") {
 			console.log("A bot is the part owner");
-			await ticketTimelineEntryCreate(ticketID, BOT_PART_OWNER_NOTIF, snap_in_token);
+			await ticketTimelineEntryCreate(ticketID, BOT_PART_OWNER_NOTIF, snap_in_token, api_base);
```
Comment (codeon) [inline 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/index.ts:39]: lets be consistent with variable name casing. snake v/s camel case.

File: 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/utils/devrev-utils.ts [hunk 4]
```diff
@@ -32,13 +36,15 @@ export async function getPartOwnersString(partObject: any) {
...
+    })
     let payload: any = {
         object: ticketID,
         type: "timeline_comment",
         body: body,
     }
-    await postCall(endpoint, payload, token);
-}
+    await devrevSDK.timelineEntriesCreate(payload);
+}
```
Comment (codeon) [inline 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/utils/devrev-utils.ts:50]: what is the benefit of this over direct SDK usage?

File: 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/utils/devrev-utils.ts [hunk 5]
```diff
@@ -2,20 +2,24 @@
  * Copyright (c) 2023 DevRev, Inc. All rights reserved.
  */
 
-import { postCall, getCall, generateQueryString } from "./api-utils"
+import { client } from "@devrev/typescript-sdk";
 
 const DEVREV_API_BASE = "https://api.devrev.ai/";
```
Comment (codeon) [inline 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/utils/devrev-utils.ts:7]: seems unused.

Comment (altafDevRev) [reply 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/utils/devrev-utils.ts:7]: Removed.

File: 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/utils/devrev-utils.ts (hunk 4 above)
Comment (altafDevRev) [reply 2-notify-owner-on-ticket-to-prod-assist/ticket-status-prod-assist/code/src/functions/ticket_stage_change/utils/devrev-utils.ts:48]: What does direct SDK usage mean?
Do you mean, directly using the SDK in place of the function - `ticketTimelineEntryCreate` ? This is just done for the separation of concern.

**Pull request closed without merging**